### ISSUE TITLE
Wrong color

### ISSUE THREAD

**Issue by pragavame** [google/model-viewer#3906]
Hey! I really need help...

![2022-10-26_05-30-47](https://user-images.githubusercontent.com/99244643/197921259-f827fd29-7928-4127-a5b8-bb77e171515a.png)

We see that everything is correct with the color.
But when I add the model to myself, my color is completely different... 

![2022-10-26_05-33-19](https://user-images.githubusercontent.com/99244643/197921504-3c664c7c-519e-485e-8603-3757b068ebd7.png)

`    <model-viewer oncontextmenu="return false;" style="width: 100%; height: 550px; margin-bottom: 50px" id="color" camera-controls  disable-pan src="models/my_model.glb" shadow-intensity="1" auto-rotate alt="3d model">
    </model-viewer>`

`      

js

```
const modelViewerParameters = document.querySelector("model-viewer#color");

      modelViewerParameters.addEventListener("load", (ev) => {
          let material = modelViewerParameters.model.materials[0];

          material.pbrMetallicRoughness.setMetallicFactor(0.5);
          material.pbrMetallicRoughness.setRoughnessFactor(1);
          
      });

      const modelViewerColor = document.querySelector("model-viewer#color");

      $("#color__clother").on("input", function() {
          let get_color = $(this).val();
          const colorString = get_color;

          if (!colorString) {
            return;
          }

          let color = colorString.replace(/[^0-9,]/g, '');

          color = color.split(',');

          color[0] = color[0] / 255;
          color[1] = color[1] / 255;
          color[2] = color[2] / 255;
         
          const [material] = modelViewerColor.model.materials;

          material.pbrMetallicRoughness.setBaseColorFactor(color);

          console.log(material);

      });
```

Whats wrong? :(

**Comment by elalish** [comment]
I'm having some trouble following - can you list out the steps of what you did?

Wild guess is that instead of exporting this modified GLB, you manually applied the RGB color you chose in your app? If that's true, then the problem is that the `baseColorFactor` in glTF is linear, not sRGB (like the texture). So if you pick a color in sRGB colorspace (what we're all used to), then you'll need to apply an sRGBToLinear conversion. We should probably add this to our documentation.

**Comment by pragavame** [comment]
> I'm having some trouble following - can you list out the steps of what you did?
> 
> Wild guess is that instead of exporting this modified GLB, you manually applied the RGB color you chose in your app? If that's true, then the problem is that the `baseColorFactor` in glTF is linear, not sRGB (like the texture). So if you pick a color in sRGB colorspace (what we're all used to), then you'll need to apply an sRGBToLinear conversion. We should probably add this to our documentation.

If I have to convert a color in RGB to SRGB, how should I do it correctly?
I divide the selected colors by 255

**Comment by pragavame** [comment]
I have a 3D model and the main function is the ability to change the color of the model.

- On my site, I change the color of it, but the color is displayed incorrectly.
- I uploaded my model to your editor and in it the color of my model is displayed correctly as I need.

I do not understand what is my mistake, that my color is different.

**Comment by elalish** [comment]
You're missing this function: https://threejs.org/docs/?q=colo#api/en/math/Color.convertSRGBToLinear

Perhaps we should make this conversion part of the MV API to simplify this step.

**Comment by pragavame** [comment]
```
      color[0] = color[0] / 255;
          color[1] = color[1] / 255;
          color[2] = color[2] / 255;
```


`material.pbrMetallicRoughness.convertSRGBToLinear(color);`


convertSRGBToLinear is not a function

**Comment by elalish** [comment]
https://github.com/mrdoob/three.js/blob/5c94e668c50d647b46f515eab156f07fec873bbe/src/math/ColorManagement.js#L3

**Comment by BenSztukowski** [comment]
Hey elalish; Model viewer had problems with adding variants previously, so we worked around it by: entering the hex color values we wanted to use as the base color factor for a model into the model-viewer editor, downloading the model, pulling it back into the editor and using the search feature to find the converted rgb value that model viewer ultimately converted it to, saved those values, then used a separate program to add variants, then exported as a .gltf, edited the RGB values under the baseColorFactor, <img width="673" alt="Screen Shot 2022-11-28 at 3 07 10 PM" src="https://user-images.githubusercontent.com/44385579/204381554-b534531b-4081-47ee-9364-6018217e751a.png">
converted back to .glb, and switched between variants to achieve color changes on our models. The colors always displayed correctly before, but now the same RGB values are producing a different hex value (and therefore, incorrect color) in the model-viewer editor. I can assume this is due to the newest update. 

I studied your answers on this board, but I can't figure out what has changed. Could you direct me to some help or shed light on why it is behaving differently?

**Comment by elalish** [comment]
The main thing was a bug in the editor we fixed, #3905. Basically the hex value displayed in the editor was not actually the value being applied to the model, since it wasn't being converted from sRGB to linear. You may have a similar issue in your own process: glTF baseColorFactor is linear, which means the conversion from Hex (sRGB) is not as simple as 1/255. 

I believe all the variant and sRGB issues are now addressed in the editor, so hopefully you can just add variants and set their base colors in the editor and export and it should all just work now. If not, please file a bug!
